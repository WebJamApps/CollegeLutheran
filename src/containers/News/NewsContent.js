import React from 'react';
import PropTypes from 'prop-types';
import ELCALogo from '../../components/elcaLogo';

const pageName = 'news';

const NewsContent = ({ books }) => (
  <div className="page-content">
    <div className="container-fluid">
      <p style={{ fontSize: '4pt', margin: '0' }}>&nbsp;</p>
      <div
        className="news-div"
        style={{
          margin: 'auto',
          textAlign: 'center',
          padding: '0',
          marginLeft: '-8px',
        }}
      >
        <iframe
          style={{
            width: '100%',
            height: '540px',
            border: '1px solid #d3d3d3',
          }}
          className="newsIFrame"
          src="https://visitor.r20.constantcontact.com/d.jsp?
            llr=9mgiltbab&p=oi&m=1101234706842&sit=ah8uvihbb&f=f26b8fbd-329c-49ba-b2c2-68eee26dbef1"
          title="News Content"
        >
          <p>Your browser does not support iframe.</p>
        </iframe>
      </div>
      <div>
        {books.length > 0 ? (
          <div className="forumsTable">
            <h3
              style={{
                paddingTop: '22px',
                marginBottom: '8px',
                fontWeight: '100',
                textAlign: 'center',
              }}
            >
              Monthly Forums
            </h3>
            <div className="table-responsive">
              <table className="newsTable">
                <thead>
                  <tr>
                    <th style={{ padding: '8px', width: '230px' }}>Title</th>
                    <th style={{ padding: '8px', width: '105px' }}>Date</th>
                  </tr>
                </thead>
                <tbody>
                  {books.map((d) => (
                    <tr key={d._id}>
                      <td className="newsUrl">
                        <a rel="noopener noreferrer" target="_blank" href={d.url}>{d.title}</a>
                      </td>
                      <td>{d.created_at.split('T')[0]}</td>
                    </tr>
                  ))}
                </tbody>
              </table>
            </div>
          </div>
        ) : (
          <p>There are currently no pdfs links to display</p>
        )}
      </div>
    </div>
    <p>&nbsp;</p>
    <p>&nbsp;</p>
    <ELCALogo pageName={pageName} />
  </div>
);

NewsContent.defaultProps = { books: [] };
NewsContent.propTypes = {
  books: PropTypes.arrayOf(
    PropTypes.shape({
      comments: PropTypes.string,
    }),
  ),
};

export default NewsContent;
