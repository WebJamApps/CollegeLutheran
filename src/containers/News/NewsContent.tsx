import React from 'react';
import ELCALogo from '../../components/elcaLogo';
import { Ibook } from '../../redux/mapStoreToProps';

interface NewsContentProps {
  books?: Ibook[];
}

const pageName = 'news';

const NewsContent = ({ books }: NewsContentProps): JSX.Element => (
  <div className="page-content">
    <div>
      <div>
        {books && books.length > 0 ? (
          <div className="forumsTable">
            <h3
              style={{
                paddingTop: '22px',
                marginBottom: '8px',
                textAlign: 'center',
              }}
            >
              Announcements
            </h3>
            <div className="TableStyle-root">
              <div className="TableContainer-root table-responsive">
                <table className="TableHeader newsTable" aria-label="News Anouncement Table">
                  <thead className="TableHead-root">
                    <tr className="TableRow-root">
                      <th className="TableCell-root TableCell-head StickyHeader newsTitle">Title</th>
                      <th className="TableCell-root TableCell-head StickyHeader" style={{ width: '103px' }}>Date</th>
                    </tr>
                  </thead>
                  <tbody className="TableBody-root">
                    {books && books.map(
                      (d: { _id: string | number | undefined; url?: string | undefined; title: React.ReactNode; created_at: string; }) => (
                        <tr className="TableRow-root" key={d._id}>
                          <td className="TableCell-root TableCell-body newsUrl">
                            <a rel="noopener noreferrer" target="_blank" href={d.url}>{d.title}</a>
                          </td>
                          <td className="TableCell-root TableCell-body">{d.created_at.split('T')[0]}</td>
                        </tr>
                      ),
                    )}
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        ) : (
          <p>There are currently no pdfs links to display</p>
        )}
      </div>
      <p style={{ fontSize: '4pt', margin: '0' }}>&nbsp;</p>
      <hr style={{ margin: '0px' }} />
      <div
        className="news-div"
        style={{
          margin: '0px 0px auto -8px',
          textAlign: 'center',
          padding: '0',
          width: '108%',
          overflow: 'hidden',
        }}
      >
        <iframe
          className="newsIFrame"
          style={{
            overflow: 'hidden',
            padding: 0,
            width: '100%',
            height: '540px',
            border: 'none',
          }}
          src="https://visitor.r20.constantcontact.com/d.jsp?
            llr=9mgiltbab&p=oi&m=1101234706842&sit=ah8uvihbb&f=f26b8fbd-329c-49ba-b2c2-68eee26dbef1"
          title="News Content"
        >
          <p>Your browser does not support iframe.</p>
        </iframe>
      </div>
    </div>
    <ELCALogo pageName={pageName} />
  </div>
);

NewsContent.defaultProps = { books: [] };

export default NewsContent;
