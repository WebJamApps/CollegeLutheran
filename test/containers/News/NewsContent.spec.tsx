import renderer from 'react-test-renderer';
import NewsContent from 'src/containers/News/NewsContent';

describe('NewsContent', () => {
  it('renders the news page', () => {
    const result: any = renderer.create(<NewsContent />).toJSON();
    expect(result).toMatchSnapshot();
  });
  it('renders the news table', () => {
    const props = { title: 'title', type: 'type', _id: 'a' };
    const result: any = renderer.create(<NewsContent books={[props]} />).toJSON();
    expect(result.children[0].children[0].children[0].props.className).toBe('forumsTable');
  });
});
