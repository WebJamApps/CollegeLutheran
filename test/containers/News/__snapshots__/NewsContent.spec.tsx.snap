// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContent renders the news page 1`] = `
<div
  className="page-content"
>
  <div>
    <div>
      <p>
        There are currently no pdfs links to display
      </p>
    </div>
    <p
      style={
        {
          "fontSize": "4pt",
          "margin": "0",
        }
      }
    >
      Â 
    </p>
    <hr
      style={
        {
          "margin": "0px",
        }
      }
    />
    <div
      className="ctct-inline-form"
      data-form-id="99081bd2-b1a5-48cd-bb60-8c9aba82c2a4"
    />
  </div>
  <div
    style={
      {
        "marginLeft": "-9px",
        "textAlign": "center",
      }
    }
  >
    <a
      href="http://www.elca.org/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="ELCA LOGO"
        id="elcaLogo"
        src="https://dl.dropboxusercontent.com/s/wkzubcmmm3pqst4/elca-logo.png?dl=0"
        style={
          {
            "margin": "auto",
            "marginLeft": "",
            "paddingTop": "30px",
            "width": "308px",
          }
        }
      />
    </a>
  </div>
</div>
`;
